testsuite/MDAnalysisTests/formats/test_libmdaxdr.py

```diff
@@ -21,376 +21,469 @@
+        big_offset = 2**32 + 1000
         # It should be safe to leave the read head so far away
         # because the file is open read-only.
+        reader._bytes_seek(big_offset)
+        assert reader._bytes_tell() == big_offset
 
 
-class Test_XTCFile(XDRFormatBaseTest):
-    xdrfile = XTCFile
-    single_frame = XTC_single_frame
-    multi_frame = XTC_multi_frame
-    offsets = np.array([0, 104, 208, 312, 416, 520, 624, 728, 832, 936])
+@pytest.mark.parametrize("xdrfile, fname", ((XTCFile, XTC_multi_frame),
+                                            (TRRFile, TRR_multi_frame)))
+def test_steps(xdrfile, fname):
+    with xdrfile(fname) as f:
+        for i, frame in enumerate(f):
+            assert frame.step == i
 
 
-def test_xtc():
-    f = XTCFile(XTC_single_frame)
-    assert_equal(f.n_atoms, 10)
-    xyz, box, step, time, prec = f.read()
-    # xtc only saves with 3 decimal places precision
-    assert_array_almost_equal(xyz.flat, np.arange(30), decimal=3)
-    assert_array_almost_equal(box, np.eye(3) * 20, decimal=3)
-    assert_equal(step, 0)
-    assert_equal(time, 10.0)
-    assert_equal(prec, 1000.0)
+@pytest.mark.parametrize("xdrfile, fname", ((XTCFile, XTC_multi_frame),
+                                            (TRRFile, TRR_multi_frame)))
+def test_time(xdrfile, fname):
+    with xdrfile(fname) as f:
+        for i, frame in enumerate(f):
+            assert frame.time == i * .5
 
 
-def test_read_multi_frame_xtc():
-    with XTCFile(XTC_multi_frame) as f:
-        f.seek(5)
-        assert_equal(f.tell(), 5)
-        assert_equal(len(f), 10)
-        assert_equal(f.tell(), 5)
-        f.seek(0)
-        assert_equal(f.tell(), 0)
-        ones = np.ones(30).reshape(10, 3)
-        box_compare = np.eye(3) * 20
-        for i, frame in enumerate(f):
-            xyz, box, step, time, prec = frame
-            assert_array_almost_equal(xyz, ones * i, decimal=3)
-            assert_array_almost_equal(box, box_compare, decimal=3)
-            assert_equal(step, i)
-            assert_equal(time, i * 0.5)
-            assert_equal(prec, 1000.0)
-
-
-@run_in_tempdir()
-def test_write_xtc():
-    with XTCFile(XTC_multi_frame) as f_in, XTCFile('foo.xtc', 'w') as f_out:
-        assert_equal(f_out.n_atoms, 0)
-        for frame in f_in:
-            f_out.write(*frame)
-
-    with XTCFile('foo.xtc') as f:
-        assert_equal(len(f), 10)
-        ones = np.ones(30).reshape(10, 3)
-        box_compare = np.eye(3) * 20
-        for i, frame in enumerate(f):
-            xyz, box, step, time, prec = frame
-            assert_array_almost_equal(xyz, ones * i, decimal=3)
-            assert_array_almost_equal(box, box_compare, decimal=3)
-            assert_equal(step, i)
-            assert_equal(time, i * 0.5)
-            assert_equal(prec, 1000.0)
+def test_box_xtc(xtc):
+    box = np.eye(3) * 20
+    for frame in xtc:
+        assert_array_almost_equal(frame.box, box, decimal=3)
```

Comment:
So here I decided to go with special functions since the API of XTC and TRR are different.